pylib_3.9.7/df/AntibodyStructurePrediction.py

```diff
@@ -0,0 +1,223 @@
+import base64
+import os
+import gzip
+from typing import Optional
+
+from Bio.SeqRecord import SeqRecord
+
+from df.bio_helper import column_to_sequences, string_to_sequence
+from df.data_transfer import ColumnData, TableData, DataFunctionRequest, DataFunctionResponse, DataFunction, DataType, \
+                             string_input_field, boolean_input_field, integer_input_field, input_field_to_column, \
+                             Notification, NotificationLevel
+
+from ruse.bio.antibody import align_antibody_sequences, ANTIBODY_NUMBERING_COLUMN_PROPERTY
+from ruse.bio.bio_data_table_helper import sequence_to_genbank_base64_str
+
+from ImmuneBuilder import ABodyBuilder2
```

Comment:
The immune builder imports will fail with the regular GlysadePython.  Unless we are planning to include ImmuneBuilder in GlysadePython we should wrap these import with

```python
try:
     from ImmuneBuilder ..
except ImportError:
     pass
````

Then in the antibody_numbering function check sys.modules and throw a message to the affect that them mab python needs to be installed

Reply:
Will it fail before that because the DF .yaml has `serviceUri: glysade.python.mab` ?  If the GlysadePython_mAb environment isn't there, would it ever make it this far?

Reply:
That's easy enough to refactor here.  I guess the reverse is also true, though - if someone had `glysade.python.mab` in the .yaml, and called another function that was dependent on `glysade.python`.  🤔

Reply:
That's true, but really unlikely.  I get your point.  Let's not clutter the code here by making a change which is probably never going to be hit.

Reply:
Let's chat about this again.  I see your point about insulating each Python environment from the other, and your suggestion isn't much clutter.  I'm wondering if there's a way to do this for each Python environment without having to do a lot of refactoring in GlysadePython.  For GlysadePython_mAb, seems simple enough.